Assets/MRTK/Core/Services/MixedRealityToolkit.cs

```diff
@@ -88,14 +93,38 @@ public MixedRealityToolkitConfigurationProfile ActiveProfile
             }
             set
             {
-                ResetConfiguration(value);
+                if(Application.isPlaying && activeProfile != null && value != null && activeProfile != value)
+                {
+                    newProfile = value;
+                }
+                else
+                {
+                    ResetConfiguration(value);
```

Comment:
Do you know why in the else case were and synchronously resetting things within this call, but for the new thing we do that async on the next update?

It does feel a bit weird that we have a divergence of behavior between these two (it feels like we should be consistent with when we chose to tear things down and restart things).

Reply:
Suggest adding detailed comment describing the scenarios.